### ISSUE TITLE
Failed to activate the nuclide package

### ISSUE THREAD

**Issue by tp** [facebook/nuclide#313]
[Enter steps to reproduce below:]
1. Tried installing Nuclide 0.111.0 from within Atom

**Atom Version**: 1.3.3
**System**: Mac OS X 10.11.2
**Thrown From**: [nuclide](https://github.com/facebook/nuclide) package, v0.111.0
### Stack Trace

Failed to activate the nuclide package

```
At "nuclide-blame" exported a "config"

Error: "nuclide-blame" exported a "config"
    at /Users/timm/.atom/packages/nuclide/lib/main.js:203:17
    at Array.forEach (native)
    at Object.activate (/Users/timm/.atom/packages/nuclide/lib/main.js:196:27)
    at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:170:19)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:150:32
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:143:26
    at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:140:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:517:21)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:444:29
    at exit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
    at triggerExitCallback (/Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:213:47)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:227:18
    at Socket.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:98:18)
    at emitOne (events.js:82:20)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```
### Commands

```
  2x -2:59.3.0 core:close (atom-text-editor.editor.is-focused)
     -2:54.8.0 settings-view:open (ol.results-view.list-tree.focusable-panel.has-collapsable-children)
     -1:45.5.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -1:41.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "web-browser",
      "tab-history",
      "autocomplete-snippets",
      "linter-jsxhint",
      "jscs-fixer",
      "linter-jshint"
    ]
  },
  "nuclide": {}
}
```
### Installed Packages

``` coffee
# User
atom-beautify, v0.28.19
atom-handlebars, v1.3.0
blame, v0.9.0
go-plus, v3.5.2
go-rename, v0.3.0
language-rust, v0.4.5
language-swift, v0.4.0
last-cursor-position, v0.9.0
linter, v1.11.3
linter-eslint, v5.2.7
linter-flow, v4.1.0
linter-js-yaml, v1.2.5
linter-jsonlint, v1.1.4
linter-less, v2.3.3
markdown-scroll-sync, v2.0.3
nuclide, v0.111.0
racer, v0.17.2
react, v0.12.10
tab-smart-sort, v0.1.4

# Dev
No dev packages
```

**Comment by nmote** [comment]
One thing that stands out is the fact that you have a package named `blame`. I wonder if this is triggering a weird issue with our unified package code that loads the individual Atom "packages." Can you try removing the `blame` package and checking whether this issue persists?

**Comment by ssorallen** [comment]
@nmote, good catch. That does appear to be the issue. The 'nuclide-blame' package's directory name is actually 'blame', which is the cause. The Nuclide loader will have to work around Atom's PackageManager. cc @zertosh

**Comment by zertosh** [comment]
This is the same issue as with the internal package formerly known as `fb-snippets`. Atom assumes in several places that the package directory name is the package name. Working around it at the loader level would only hide the problem. The real solution is to rename the directories of features to their `package.json` name – aka flatten the directory structure.

**Comment by nmote** [comment]
Makes sense. Assuming it's not realistic to remove that assumption from Atom, this is a pretty convincing argument for flattening our directory structure.

**Comment by ssorallen** [comment]
The blob from [L71-111](https://github.com/facebook/nuclide/blob/00854c41ac85734ce78617ca6f15b0c6a31ba540/lib/main.js#L71-L111) could go away.

@zertosh Will this still be a problem if, say, someone else publishes a 'nuclide-blame' package in the future? We will no longer own the names of Nuclide's "packages", so a collision is likely to occur. Do we still need to reserve the names of all of Nuclide's features-cum-packages?

**Comment by zertosh** [comment]
@ssorallen That would be a problem. We'll have to ask the Atom team if we can publish, then unpublish, a package to de facto _blacklist_ the name. Otherwise, the `Settings -> Install` view would remain crazy crowded (like it is now - before we unpublish).

**Comment by ssorallen** [comment]
The settings-view's definition of "installed" could be changed to at least address the Settings page. It currently only checks `atom.packages.isPackageLoaded`: https://github.com/atom/settings-view/blob/2f4ad155e2159f9ce3e5242ab711ca8a5d9c25a7/lib/package-manager.coffee#L25-L31 Although maybe that's catering to Nuclide's overloading of a "package".

**Comment by zertosh** [comment]
We caught this same issue internally with someone running [file-watcher](https://atom.io/packages/file-watcher). The plan is to flatten `/pkg` so each directory is the actual feature name. Since I don't think `hg` will rebase file moves, we'll have to see how many outstanding diffs there are to figure when the least disruptive time is for this codemod.

**Comment by raelmiu** [comment]
So is there any way to use nuclide until this fix happens?

**Comment by zertosh** [comment]
The fix should come out this week. The work around is to deactivate the package whose name is conflicting. In the case of the OP, he must have a package called "blame" that once deactivated allows nuclide to start. Whatever the error message is, the name after the "nuclide-" is the conflicting package.

**Comment by creativeprogramming** [comment]
+1

**Comment by tychota** [comment]
**Not sure it is related but atom does link it to this issue**

------------------------

**Atom Version**: 1.5.0-beta0
**System**: Unknown Windows Version
**Thrown From**: [nuclide](https://github.com/facebook/nuclide) package, v0.112.0


### Stack Trace

Failed to activate the nuclide package

```
At Cannot read property 'mainModule' of null

TypeError: Cannot read property 'mainModule' of null
  at C:\Users\Salas\.atom\packages\nuclide\lib\main.js:201:34
  at Array.forEach (native)
  at Object.activate (C:\Users\Salas\.atom\packages\nuclide\lib\main.js:196:27)
  at Package.module.exports.Package.activateNow (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\package.js:183:20)
  at C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\package.js:156:32
  at Package.module.exports.Package.measure (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\package.js:92:15)
  at C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\package.js:149:26
  at Package.module.exports.Package.activate (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\package.js:146:34)
  at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\package-manager.js:526:21)
  at C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:385:60
  at exit (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:73:16)
  at triggerExitCallback (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\buffered-process.js:213:47)
  at ChildProcess.<anonymous> (C:\Users\Salas\AppData\Local\atom\app-1.5.0-beta0\resources\app.asar\src\buffered-process.js:235:18)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)

```

### Commands

```
     -8:59.3.0 window:toggle-dev-tools (a.about-atom-io)
     -7:37.2.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-seti-syntax.theme-seti-ui)
     -6:30.4.0 about:view-release-notes (atom-workspace.workspace.scrollbars-visible-always.theme-seti-syntax.theme-seti-ui)
```

### Config

```json
{
  "core": {
    "themes": [
      "seti-ui",
      "seti-syntax"
    ]
  },
  "nuclide": {
    "installRecommendedPackages": true,
    "nuclide-arcanist": {},
    "nuclide-buck-files": {},
    "nuclide-debugger-hhvm": {},
    "nuclide-diagnostics-ui": {
      "enableAutofix": true
    },
    "nuclide-file-tree": {
      "revealFileOnSwitch": true
    },
    "nuclide-flow": {
      "diagnosticsOnFly": true
    },
    "nuclide-format-js": {
      "reprint": true
    },
    "nuclide-fuzzy-filename-provider": {},
    "nuclide-health": {
      "showActiveHandles": true,
      "showActiveRequests": true,
      "showCpu": true,
      "showHeap": true,
      "showKeyLatency": true,
      "showMemory": true
    },
    "nuclide-home": {
      "showHome": false
    },
    "nuclide-objc": {},
    "nuclide-ocaml": {},
    "use": {
      "nuclide-clang-atom": false,
      "nuclide-file-tree": false
    }
  }
}
```

### Installed Packages

```coffee
# User
color-picker, v2.1.0
haskell-grammar, v0.4.0
highlight-selected, v0.11.1
language-babel, v2.14.1
language-ini, v1.14.0
language-ocaml, v1.1.2
language-thrift, v1.0.2
minimap, v4.19.0
nuclide, v0.112.0
pigments, v0.21.0
seti-syntax, v0.4.1
seti-ui, v0.8.1
tool-bar, v0.1.11

# Dev
No dev packages
```

**Comment by samuelcole** [comment]
I got that `mainModule` error as well.

**Comment by julienmarie** [comment]
Same here
`Cannot read property 'mainModule' of null`

**Comment by guyca** [comment]
here as well

**Comment by hufeng** [comment]
I got that mainModule error as well.

**Comment by ikrauchanka** [comment]
I got that mainModule error as well.

**Comment by zertosh** [comment]
I apologize for the hold up on this. The workaround is to disable whatever package you have installed that matches one of these directory names https://github.com/facebook/nuclide/tree/master/pkg/nuclide. Then if you don't want that particular Nuclide feature, disable it from Settings -> Packages -> nuclide -> Settings -> 'Enable the "xxx" feature', then re-enable the package that clashed with the name.

The real fix for this won't be in tomorrows release, but may be in the following one.

**Comment by mmallorquin** [comment]
I got this error when I open nuclide

> Failed

 to activate the nuclide package
Cannot find module '../pkg/nuclide/feature-config'

<

**Comment by mmallorquin** [comment]
I got this error when I open nuclide on linux elementary OS

>  Failed
>  to activate the nuclide package
> Cannot find module '../pkg/nuclide/feature-config'

**Comment by zertosh** [comment]
The fix is in 2057807, we're aiming for a release tomorrow. Sorry for the long wait on this. As you can see, the change was not trivial.

**Comment by rulatir** [comment]
Just updated to 127, got this immediately after update, and got sent here by the error box.

**Atom Version**: 1.6.0
**System**: linux 4.4.5-1-ARCH
**Thrown From**: [nuclide](https://github.com/facebook/nuclide) package, v0.127.0


### Stack Trace

Failed to activate the nuclide package

```
At Cannot read property 'mainModule' of null

TypeError: Cannot read property 'mainModule' of null
  at /fasthome/rulatir/.atom/packages/nuclide/lib/main.js:204:34
  at Array.map (native)
  at Object.activate (/fasthome/rulatir/.atom/packages/nuclide/lib/main.js:196:40)
  at Package.module.exports.Package.activateNow (/usr/share/atom/resources/app.asar/src/package.js:181:20)
  at /usr/share/atom/resources/app.asar/src/package.js:155:32
  at Package.module.exports.Package.measure (/usr/share/atom/resources/app.asar/src/package.js:92:15)
  at /usr/share/atom/resources/app.asar/src/package.js:148:26
  at Package.module.exports.Package.activate (/usr/share/atom/resources/app.asar/src/package.js:145:34)
  at PackageManager.module.exports.PackageManager.activatePackage (/usr/share/atom/resources/app.asar/src/package-manager.js:526:21)
  at /usr/share/atom/resources/app.asar/node_modules/settings-view/lib/package-manager.js:385:60
  at exit (/usr/share/atom/resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
  at triggerExitCallback (/usr/share/atom/resources/app.asar/src/buffered-process.js:213:47)
  at /usr/share/atom/resources/app.asar/src/buffered-process.js:227:18
  at Socket.<anonymous> (/usr/share/atom/resources/app.asar/src/buffered-process.js:98:18)
  at emitOne (events.js:82:20)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:469:12)

```

### Commands

```
     -2:04.2.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "tree-view"
    ]
  },
  "nuclide": {
    "installRecommendedPackages": true,
    "nuclide-arcanist": {},
    "nuclide-buck-files": {},
    "nuclide-clang-atom": {},
    "nuclide-datatip": {},
    "nuclide-debugger-hhvm": {},
    "nuclide-flow": {},
    "nuclide-format-js": {},
    "nuclide-hack": {},
    "nuclide-health": {},
    "nuclide-home": {
      "showHome": false
    },
    "nuclide-ios-simulator-logs": {},
    "nuclide-ocaml": {},
    "nuclide-output": {},
    "nuclide-react-native": {},
    "nuclide-working-sets": {}
  }
}
```

### Installed Packages

```coffee
# User
haskell-grammar, v0.4.0
highlight-selected, v0.11.2
language-babel, v2.16.0
language-ini, v1.15.0
language-ocaml, v1.1.2
language-thrift, v1.0.2
minimap, v4.21.0
nuclide, v0.127.0
symbols-tree-view, v0.11.0
tool-bar, v0.4.0

# Dev
No dev packages
```

**Comment by zertosh** [comment]
@rulatir I'm seeing that after the update is complete. Reloading Atom seems to make everything work again. Have you tried that?

**Comment by rulatir** [comment]
@zertosh yup, reloading fixed it.

**Comment by ryanpcmcquen** [comment]
Just got hit by this with the upgrade to `0.130.0`:


**Atom Version**: 1.6.2
**System**: Mac OS X 10.11.4
**Thrown From**: [nuclide](https://github.com/facebook/nuclide) package, v0.130.0


### Stack Trace

Failed to activate the nuclide package

```
At Cannot read property 'mainModule' of null

TypeError: Cannot read property 'mainModule' of null
  at /Users/ryan.mcquen/.atom/packages/nuclide/lib/main.js:220:34
  at Array.map (native)
  at Object.activate (/Users/ryan.mcquen/.atom/packages/nuclide/lib/main.js:210:40)
  at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:181:20)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:155:32
  at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:148:26
  at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:145:34)
  at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:526:21)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:385:60
  at exit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
  at triggerExitCallback (/Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:213:47)
  at ChildProcess.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:235:18)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)

```

### Commands

```
     -3:11.4.0 atom-beautify:beautify-editor (atom-text-editor.editor.is-focused)
     -3:09.4.0 core:save (atom-text-editor.editor.is-focused)
  7x -3:07.3.0 core:move-down (atom-text-editor.editor.is-focused)
 18x -3:05.0 core:move-up (atom-text-editor.editor.is-focused)
  5x -3:04.0 core:move-down (atom-text-editor.editor.is-focused)
  8x -3:03.5.0 core:move-up (atom-text-editor.editor.is-focused)
 15x -3:02.1.0 core:move-down (atom-text-editor.editor.is-focused)
  4x -3:00.9.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -3:00.2.0 core:move-left (atom-text-editor.editor.is-focused)
     -2:59.6.0 core:move-down (atom-text-editor.editor.is-focused)
     -2:58.2.0 bracket-matcher:go-to-matching-bracket (atom-text-editor.editor.is-focused)
  8x -2:56 core:move-up (atom-text-editor.editor.is-focused)
  8x -2:50.3.0 core:move-down (atom-text-editor.editor.is-focused)
     -2:47.3.0 bracket-matcher:go-to-matching-bracket (atom-text-editor.editor.is-focused)
     -2:47.3.0 core:move-down (atom-text-editor.editor.is-focused)
     -1:26.4.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "tree-view"
    ]
  },
  "nuclide": {
    "lastConnectionDetails": {
      "lastOfficialRemoteServerCommand": "nuclide-start-server",
      "updatedConfig": {
        "authMethod": "PRIVATE_KEY",
        "cwd": "/home/ryan.mcquen/terraServers/",
        "displayTitle": "(default)",
        "host": "spidercliff",
        "password": "",
        "pathToPrivateKey": "/Users/ryan.mcquen/.ssh/id_rsa",
        "remoteServerCommand": "nuclide-start-server -p 4380",
        "sshPort": "22",
        "username": "ryan.mcquen"
      }
    },
    "nuclide-home": {
      "showHome": false
    }
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.2
linter-jslint, v10.2.0
nuclide, v0.130.0
remote-edit, v1.8.24

# Dev
No dev packages
```

Reloading _seems_ to have fixed it.

**Comment by ryanpcmcquen** [comment]
Just got hit by this with the upgrade to `0.130.0`:


**Atom Version**: 1.6.2
**System**: Mac OS X 10.11.4
**Thrown From**: [nuclide](https://github.com/facebook/nuclide) package, v0.130.0


### Stack Trace

Failed to activate the nuclide package

```
At Cannot read property 'mainModule' of null

TypeError: Cannot read property 'mainModule' of null
  at /Users/ryan.mcquen/.atom/packages/nuclide/lib/main.js:220:34
  at Array.map (native)
  at Object.activate (/Users/ryan.mcquen/.atom/packages/nuclide/lib/main.js:210:40)
  at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:181:20)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:155:32
  at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:148:26
  at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:145:34)
  at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:526:21)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:385:60
  at exit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
  at triggerExitCallback (/Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:213:47)
  at ChildProcess.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:235:18)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)

```

### Commands

```
     -3:11.4.0 atom-beautify:beautify-editor (atom-text-editor.editor.is-focused)
     -3:09.4.0 core:save (atom-text-editor.editor.is-focused)
  7x -3:07.3.0 core:move-down (atom-text-editor.editor.is-focused)
 18x -3:05.0 core:move-up (atom-text-editor.editor.is-focused)
  5x -3:04.0 core:move-down (atom-text-editor.editor.is-focused)
  8x -3:03.5.0 core:move-up (atom-text-editor.editor.is-focused)
 15x -3:02.1.0 core:move-down (atom-text-editor.editor.is-focused)
  4x -3:00.9.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -3:00.2.0 core:move-left (atom-text-editor.editor.is-focused)
     -2:59.6.0 core:move-down (atom-text-editor.editor.is-focused)
     -2:58.2.0 bracket-matcher:go-to-matching-bracket (atom-text-editor.editor.is-focused)
  8x -2:56 core:move-up (atom-text-editor.editor.is-focused)
  8x -2:50.3.0 core:move-down (atom-text-editor.editor.is-focused)
     -2:47.3.0 bracket-matcher:go-to-matching-bracket (atom-text-editor.editor.is-focused)
     -2:47.3.0 core:move-down (atom-text-editor.editor.is-focused)
     -1:26.4.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "tree-view"
    ]
  },
  "nuclide": {
    "lastConnectionDetails": {
      "lastOfficialRemoteServerCommand": "nuclide-start-server",
      "updatedConfig": {
        "authMethod": "PRIVATE_KEY",
        "cwd": "",
        "displayTitle": "(default)",
        "host": "",
        "password": "",
        "pathToPrivateKey": "",
        "remoteServerCommand": "",
        "sshPort": "",
        "username": ""
      }
    },
    "nuclide-home": {
      "showHome": false
    }
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.2
linter-jslint, v10.2.0
nuclide, v0.130.0
remote-edit, v1.8.24

# Dev
No dev packages
```

Reloading _seems_ to have fixed it.

**Comment by zertosh** [comment]
This can happen for multiple reasons. The reason for it happening immediately after an upgrade, but not after restarting Atom, is because when Atom updates a package, it doesn't actually clear the require cache. So unless very specific conditions are met, you end up loading the same modules as pre-update from memory but with a different disk layout.

The solution we're going to implement, which is similar to VS Code's, is to require a restarting the editor after an upgrade. In the case of Nuclide, we won't force you to reload, but the update will refuse to load until you've restarted (showing you a notification of course).

**Comment by montogeek** [comment]
Restart everything time please :)

**Comment by jhohlfeld** [comment]
Got this error after updating to 0.141.0 today. The error was not related to a particular sub-module but rather nuclide in general. 

After reading this thread, I finally restarted Atom. Once up again, all interface icons in Atom are gone ;( and nuclide complains about flow not being installed.

**Comment by nmote** [comment]
I recently added a popup that appears once per Nuclide run if you have a project with a .flowconfig but the Flow binary cannot be located, but that went out two weeks ago. In any case it's expected behavior now, and shows that at least some of Nuclide is working. For the other issue, I don't know if it's a Nuclide problem but it may be worth opening another issue with a screenshot.

**Comment by zertosh** [comment]
@jhohlfeld- you mean you don't see a toolbar? That's odd. Please open a new issue.

**Comment by jhohlfeld** [comment]
@zertosh @nmote Sorry, I can't reproduce and take screenshots anymore. I just removed Atom completely and installed a fresh nuclide package just now.

However, I submit a new issue regarding Flow not being installed: #537 
I think this is still worth looking at.
